app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -42,21 +41,36 @@ protected DateTime getScheduledTimeBasedOnEvent(ScheduleContext context) {
     protected void addScheduledActivityForAllTimes(List<ScheduledActivity> scheduledActivities, 
```

Comment:
I've refactored this so the method takes LocalDate and LocalTime. it's clearer now that in all cases where times have not been set, we're using MIDNIGHT.